WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -2655,6 +2690,160 @@ namespace Plugin {
             }
         }
 
+#if HAS_SCREEN_HDR_API
+        WPEFramework::PluginHost::IPlugin* QueryDisplayInfoPlugin()
+        {
+            if (_displayInfoPlugin) {
+                return _displayInfoPlugin;
+            }
+            if (!_service) {
+                return nullptr;
+            }
+            _displayInfoPlugin = _service->QueryInterfaceByCallsign<WPEFramework::PluginHost::IPlugin>(_T("DisplayInfo"));
+            return _displayInfoPlugin;
+        }
+
+        bool SubscribeHDRCapabilities()
+        {
+            auto* displayInfoPlugin = QueryDisplayInfoPlugin();
+            if (!displayInfoPlugin || _displayConnectionProps) {
+                return false;
+            }
+            _displayConnectionProps = displayInfoPlugin->QueryInterface<Exchange::IConnectionProperties>();
+            if (!_displayConnectionProps) {
+                return false;
+            }
+            _displayConnectionProps->Register(this);
+            SYSLOG(Logging::Notification, (_T("Subscribed to HDR capabilities updates from DisplayInfo plugin.")));
+            return true;
+        }
+
+        void UnsubscribeHDRCapabilities()
+        {
+            if (_displayConnectionProps) {
+                _displayConnectionProps->Unregister(this);
+                _displayConnectionProps->Release();
+                _displayConnectionProps = nullptr;
+            }
+            _adminLock.Lock();
+            if (_hdrRefreshJob.IsValid()) {
+                Core::WorkerPool::Instance().Revoke(_hdrRefreshJob);
+                _hdrRefreshJob.Release();
+            }
+            _adminLock.Unlock();
+            if (_displayInfoPlugin) {
+                _displayInfoPlugin->Release();
+                _displayInfoPlugin = nullptr;
+            }
+            SYSLOG(Logging::Notification, (_T("Unsubscribed from HDR capabilities updates from DisplayInfo plugin.")));
+        }
+
+        void RefreshHDRSupport()
+        {
+            auto displayInfoPlugin = QueryDisplayInfoPlugin();
+            if (!displayInfoPlugin) {
+                return;
+            }
+            Exchange::IHDRProperties* hdrPropIface = displayInfoPlugin->QueryInterface<Exchange::IHDRProperties>();
+            if (!hdrPropIface) {
+                return;
```

Comment:
Probably reset _hdrSupported to false on early return. and in below eary returns too.